include/ck/utility/dynamic_buffer.hpp

```diff
@@ -241,8 +272,31 @@ struct DynamicBuffer
         {
             constexpr index_t t_per_x = scalar_per_x_vector / scalar_per_t_vector;
 
+            // To Fix Compilation Error Stream-K reduction caused by type convertion
+#if 0
             amd_buffer_store<remove_cvref_t<T>, t_per_x, coherence>(
                 x, p_data_, i, is_valid_element, element_space_size_ / PackedSize);
+#endif
+
+#if 1
+            using vector_t = typename vector_type_maker<remove_cvref_t<T>, t_per_x>::type::type;
+            vector_t tmp;
+
+            // Convert the x value to the appropriate vector type
+            if constexpr(std::is_same_v<remove_cvref_t<X>, vector_t>)
+            {
+                // If x is already the correct type, use it directly
+                tmp = x;
+            }
+            else
+            {
+                // Otherwise, use memory copy to safely convert between types
```

Comment:
Are comments really needed to explain this very straightforward code?
```suggestion
            if constexpr(std::is_same_v<remove_cvref_t<X>, vector_t>)
            {
                tmp = x;
            }
            else
            {
```